### PR TITLE
Reset zoom on time range change

### PR DESCRIPTION
**Pull request by jaredscheib**
- [x] CHANGELOG.md updated with a link to the PR (not the Issue)
  - [x] Rebased/mergable
  - [x] Tests pass
  - [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

Connect #1481 

### The problem
Graphs that had been zoomed in on, would remained zoomed on the zoomed period of time, even if a new time range was selected.

### The Solution
Reset graphs' zoom when a new time range is updated.

### Notes
- When time range is changed, the graph will synchronously call resetZoom because timeRange's prop change is synchronous, and will shortly thereafter redraw the graph when the async fetch for updated `timeSeriesData` returns. At least one idea of how alternatives to this flickering effect:
  - retain the values for `dateWindow` in a dygraph `zoomCallback` and use that `dateWindow` within the `updateOptions` call from within `Dygraph.componentDidUpdate`, so that a redraw only happens once (in conjunction with other conditional logic around timeRange) -- this would cause a slight delay between changing timeRange and seeing the graph both redraw and have its zoom reset

### CODE REVIEW

**Comment by jaredscheib** [comment]
There are some outstanding questions on the connected issue that may merit confirmation before merging this. Depends what UX we want.

File: ui/src/shared/components/Dygraph.js [hunk 1]
```diff
@@ -149,6 +153,40 @@ export default React.createClass({
...
   },
 
+  // Only update the component if props other than timeRange have updateDashboard.
+  // If only timeRange has changed, only reset the zoom.
+  // TODO: possibly try to use dateWindow and a zoomCallback to updateOptions
+  // in comonentDidUpdate so that zoom and redraw happen at the same time
+  shouldComponentUpdate(nextProps) {
+    const timeRangeChanged =
+      nextProps.timeRange.lower !== this.props.timeRange.lower ||
+      nextProps.timeRange.upper !== this.props.timeRange.upper
```
Comment (lukevmorris) [inline ui/src/shared/components/Dygraph.js]: Can this check be rolled into L172? Maybe:

1. Compile list of properties that have changed
2. If list includes `lower` or `upper`, then resetZoom
3. Remove `lower` and `upper`
4. Return length of list

**Comment by lukevmorris** [comment]
Code looks good to me -- let's test around to ensure this didn't break anything. There seems to be an issue in CEO on first pass.

**Comment by jaredscheib** [comment]
Thanks -- good ideas. Refactored per your suggestion, and then some.

Not able to repro the CEO issue.

**Comment by lukevmorris** [comment]
But it works on the first graph!

**Pull request merged**